src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -37,6 +46,18 @@ class Lambda {
       }
     }
 
+    val recordsAddedResult: Seq[UserRecordResult] = Await.result(
+      PurgedEventWriter.addRecords(rawRecords, kinesisProducer), Duration(context.getRemainingTimeInMillis, MILLISECONDS)
```

Comment:
sometimes if a kinesis stream is 'full' weird things happen when you try to add records to it - I think on dotcom we had an issue where the events got put in a queue which eventually used up lots of memory and caused problems in the app. There are some details here https://github.com/guardian/frontend/pull/19061 - hopefully we can come up with a solution that's a bit less involved than that one though!